packages/devtools_app/lib/src/shared/console/widgets/display_provider.dart

```diff
@@ -100,22 +78,47 @@ class DisplayProvider extends StatelessWidget {
             ),
           ],
         ),
-        selectionControls: _selectionControls(
-          variable: variable,
-          onInspect: serviceManager.inspectorService == null
-              ? null
-              : (delegate) => _handleInspect(delegate, context),
-        ),
-        onTap: onTap,
       ),
     );
   }
 
+  List<ContextMenuButtonItem> _getMenuButtons(
+    BuildContext context,
+  ) {
+    final menuButtons = <ContextMenuButtonItem>[];
+    if (variable.isRerootable) {
+      menuButtons.add(
+        ContextMenuButtonItem(
+          onPressed: () {
+            ContextMenuController.removeAny();
+            final ref = variable.ref;
+            serviceManager.consoleService.appendBrowsableInstance(
+              instanceRef: variable.value as InstanceRef?,
+              isolateRef: ref?.isolateRef,
+              heapSelection: ref?.heapSelection,
+            );
+          },
+          label: 'Reroot',
+        ),
+      );
+    }
+    if (serviceManager.inspectorService != null && variable.isRoot) {
+      menuButtons.add(
+        ContextMenuButtonItem(
+          onPressed: () {
+            ContextMenuController.removeAny();
+            _handleInspect(context);
+          },
+          label: 'Inspect',
+        ),
+      );
+    }
+    return menuButtons;
```

Comment:
this can be condensed:
```dart
return [
  if (variable.isRerootable)
    // reroot button
  if (serviceManager.inspectorService != null && variable.isRoot)
    // inspect button
];
```